src/messages/formatters.ts

```diff
@@ -145,6 +145,33 @@ export function hyperlink(content: string, url: string | URL, title?: string) {
 	return title ? `[${content}](${url} "${title}")` : `[${content}](${url})`;
 }
 
+/**
+ * Formats the user ID into a user mention.
+ * @param userId The user ID to format.
+ * @returns The formatted user mention.
+ */
+export function userMention<C extends string>(userId: C): `<@!${C}>` {
```

Comment:
```suggestion
export function userMention<C extends Snowflake>(userId: C): `<@!${C}>` {
```
This type, along with the other two, should extend Snowflake from discord-api-types.